tests/lib/authentication.py

```diff
@@ -28,7 +30,7 @@ def generate_authentication(env):
         else:
             raise ValueError("Unknown value: ", env)
 
-        if tokens.get(env, None) is None or (call_time is None or call_time + 595 < int(time())):
+        if env not in self.tokens or self.tokens[env][1] + 595 < int(time()):

```

Comment:
Can we make a local var of `self.tokens[env][1]` and annotate what it is? I'm still unsure of what call_time was meant to denote previously

Reply:
`call_time` was the unix timestamp when the `Authentication` class last retrieved a (read: any) token. This meant that if I was using `Authentication` to manage multiple tokens it would actually renew each one more frequently than necessary.

My change moves `call_time` into the second index of a tuple stored in `self.tokens`, so each token records when it was issued. I'll annotate this as you've requested.

Reply:
Does it not make sense to name it along the lines of `token_creation_time`, `call_time` is possibly a bit ambiguous

Reply:
Is the current implementation better?